### PR TITLE
feat: [CDS-54634]: support Nexus, Artifactory, Jenkins and Custom artifact sources

### PR DESCRIPTION
**Pull request by chetanvekariya-harness**
### Summary
- Support Nexus2, Nexus3, Artifactory, Jenkins and Custom artifact sources

#### Screenshots
Screenshot 2023-03-08 at 12 00 44 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by chetanvekariya-harness** [comment]
retrigger titlecheck

**Review by rameshx — DISMISSED** [review]

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactHelper.ts [hunk 1]
```diff
@@ -278,7 +278,15 @@ export const allowedArtifactTypes: Record<ServiceDefinition['type'], Array<Artif
...
   GoogleCloudFunctions: [ENABLED_ARTIFACT_TYPES.GoogleCloudStorage],
-  AwsLambda: [ENABLED_ARTIFACT_TYPES.AmazonS3, ENABLED_ARTIFACT_TYPES.Ecr],
+  AwsLambda: [
+    ENABLED_ARTIFACT_TYPES.AmazonS3,
+    ENABLED_ARTIFACT_TYPES.Ecr,
+    ENABLED_ARTIFACT_TYPES.Nexus2Registry,
+    ENABLED_ARTIFACT_TYPES.Nexus3Registry,
+    ENABLED_ARTIFACT_TYPES.ArtifactoryRegistry,
+    ENABLED_ARTIFACT_TYPES.Jenkins,
+    ENABLED_ARTIFACT_TYPES.CustomArtifact
```
Comment (PratyushGar) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactHelper.ts:288]: For custom artifacts, we have a FF `CUSTOM_ARTIFACT_NG`, few swimlanes consume it behind FF in `ServiceV2ArtifactsSelection.tsx` & `ArtifactsSelection.tsx`

**Review by PratyushGar — DISMISSED** [review]

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactHelper.ts (hunk 1 above)
Comment (chetanvekariya-harness) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactHelper.ts]: @PratyushGar - Made the change. 
But, any idea why SSH and WinRM swim-lanes are directly injecting Custom Artifact in the supported list without checking for FF?

Comment (PratyushGar) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactHelper.ts]: Not sure might have happened due to not being aware of FF which I also got to know few months back

**Review by PratyushGar — DISMISSED** [review]

**Review by rameshx — DISMISSED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Pull request merged**